Utilities/Services/PostDeserializer.cs

```diff
@@ -34,19 +36,29 @@ public PostDeserializer(IXMLFileValidator fVal,string dFile,string dXSD )
         public List<Post> DeserializeXML(string fileName, string xsd)
```

Comment:
I think this interface actually needs some hefty changes. Passing the filename and xsd in made some sense when it was loading it at runtime, but that's not happening anymore; it's loading it once, then keeping it around. The filename and xsd aren't actually *doing* anything at that point, if you pass in a different filename it'll just silently discard it and return the same data it had before. That's a recipe for confusion.

Recommend changing the constructor variables to not be "default", but rather to actually be the filename and XSD. Do the load itself in the constructor instead of lazily, then get rid of the parameters to DeserializeXML.